langserver/server.go

```diff
@@ -150,135 +141,30 @@ func ServerFromStream(ctx context.Context, stream jsonrpc2.Stream, config *Confi
 	return ctx, Server{s}
 }
 
-// nolint:funlen
 func (s *server) connectPrometheus(url string) error {
-	s.prometheusMu.Lock()
-	defer s.prometheusMu.Unlock()
-
-	s.PrometheusURL = ""
-	s.prometheus = nil
-
-	if strings.TrimSpace(url) == "" {
-		return nil
-	}
-
-	var err error
-
-	s.prometheus, err = api.NewClient(api.Config{Address: url})
-	err = errors.Wrapf(err, "Failed to connect to prometheus: %s\n", url)
-
-	if err == nil {
-		// nolint: errcheck
-		s.client.LogMessage(s.lifetime, &protocol.LogMessageParams{
-			Type:    protocol.Info,
-			Message: fmt.Sprint("Prometheus: ", url),
-		})
-	}
-
-	if !s.headless {
-		testurl := fmt.Sprint(url, "/api/v1/status/buildinfo")
-
-		resp, err := http.Get(testurl) // nolint: gosec
-
+	if s.prometheusClient != nil {
```

Comment:
The `prometheusClient` should never be `nil` after successful server initialisation, since that would make everything that calls methods on it crash. I think that's what causes the test failures.

Reply:
If not already present before it should be initialized in `ServerFromStream`.

Reply:
done